openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/edgefilters/ch/DownwardSearchEdgeFilter.java

```diff
@@ -20,19 +20,50 @@
 
 public class DownwardSearchEdgeFilter extends CHLevelEdgeFilter {
 	protected final BooleanEncodedValue accessEnc;
+	private boolean useCore = false;
+	private boolean useCoreTurnRestrictions = false;
+	private boolean swap = false;
+	private int coreNodeLevel = -1;
 
 
 	public DownwardSearchEdgeFilter(CHGraph g, FlagEncoder encoder) {
 		super(g, encoder);
 		accessEnc = encoder.getAccessEnc();
 	}
 
+	public DownwardSearchEdgeFilter(CHGraph g, FlagEncoder encoder, boolean useCore) {
+		this(g, encoder);
+		this.useCore = useCore;
+		this.coreNodeLevel = maxNodes + 1;
+
+	}
+
+	public DownwardSearchEdgeFilter(CHGraph g, FlagEncoder encoder, boolean useCore, boolean useCoreTurnRestrictions, boolean swap) {
+		this(g, encoder, useCore);
+		if(useCore == false && useCoreTurnRestrictions == true)
+			throw new IllegalArgumentException("If turn restrictions in core should be respected, core must be respected, too.");
+		this.swap = swap;
+	}
+
 	@Override
 	public boolean accept(EdgeIteratorState edgeIterState) {
 		int adj = edgeIterState.getAdjNode(); 
-
+		if(useCore && !useCoreTurnRestrictions) {
```

Comment:
All the if statements could be avoided if the specialized code for core matrix was moved to a dedicated subclass of `DownwardSearchEdgeFilter`. Is there any particular reason to keep it all in one class rather than splitting into across separate classes? If not, I would recommend refactoring.

Reply:
Please re-view. I removed an unnecessary if case.

Reply:
My question still remains valid: why is accommodating of two different algorithms implemented through a conditional statement rather than having distinct classes? I believe any branching points such as this one should be avoided if there are straightforward alternatives.

Reply:
I think this will make it messier, as the functionality of both filters is exactly the same, just that one is `<=` and the other is `<`. Generating a second class to accomplish this seems very much like overkill to me, as the filter are even used for the same procedure in both algorithms. I could even rename the boolean to something like `inclusive` so that the algorithm names are not present anymore. But I do not care, so I will just write a second class that has the same content.

Reply:
Done

Reply:
Thanks, I appreciate the extra effort you took in accommodating my suggestion :)